lineapy/data/graph.py

```diff
@@ -94,8 +94,13 @@ def visit_order(self) -> Iterator[Node]:
                     if parent_id in self.ids
                 ]
             )
-            # First we add all of the nodes to the queue which have no parents
 
+            # Removing certain edges to ensure the graph for execution is acyclic, to generate a proper order for execution of nodes
+            if isinstance(node, IfNode):
+                if node.companion_id is not None:
+                    n_remaining_parents -= 1
+
```

Comment:
you need to document this a  bit more. this will not be very clear in a few months esp because we dont touch this section too much.